utils_nlp/models/transformers/common.py

```diff
@@ -81,99 +81,86 @@ def set_seed(seed, cuda=True):
         if cuda and torch.cuda.is_available():
             torch.cuda.manual_seed_all(seed)
 
+    @staticmethod
+    def get_default_optimizer(model, weight_decay, learning_rate, adam_epsilon):
+        no_decay = ["bias", "LayerNorm.weight"]
+        optimizer_grouped_parameters = [
+            {
+                "params": [
+                    p for n, p in model.named_parameters() if not any(nd in n for nd in no_decay)
+                ],
+                "weight_decay": weight_decay,
+            },
+            {
+                "params": [
+                    p for n, p in model.named_parameters() if any(nd in n for nd in no_decay)
+                ],
+                "weight_decay": 0.0,
+            },
+        ]
+        optimizer = AdamW(optimizer_grouped_parameters, lr=learning_rate, eps=adam_epsilon)
+        return optimizer
+
+    @staticmethod
+    def get_default_scheduler(
+        optimizer, warmup_steps, data_loader, max_steps, num_epochs, gradient_accumulation_steps
+    ):
```

Comment:
Can we move the logic to determine the max_steps out of get_default_scheduler? get_default_scheduler only need optimizer, max_steps, warmup_steps.

Reply:
we can wrap the logic of determining the max_steps as a function too.

Reply:
Yes, I actually did that last week. Let me push those.